File: jour2/p2.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#include <algorithm>
#include <sstream>
#include <map>
using namespace std;

int main() {
    fstream monFichier("input.txt", ios::in);

    string s;
    int somme = 0;

    while (getline(monFichier, s)) {
        int n = 0;
        int last = 0;
        string sens = "first";
        bool test = true;
        s += " ";
        for (char& e : s) {
            if (e == ' ') {
                if (sens == "first") {
                    sens = "None";
                } else if (sens == "None") {
                    if (n > last && abs(n - last) <= 3) {
                        sens = "Croissant";
                    } else if (n < last && abs(n - last) <= 3) {
                        sens = "Decroissant";
                    } else {
                        test = false;
                    }
                } else if (sens == "Croissant") {
                    if (n <= last) {
                        test = false;
                    } else if (abs(n - last) > 3) {
                        test = false;
                    }
                } else if (sens == "Decroissant"){
                    if (n >= last) {
                        test = false;
                    } else if (abs(n- last ) > 3) {
                        test = false;
                    }
                }
                last = n;
                n = 0;
            }
            else 
            {

                n = n * 10 + (e - '0');
            }
        }


        if (test) {
            somme ++;
        }
    }

    cout << somme << endl;
}

File: jour2/p2.py
f = open("input.txt", "r")

somme = 0

for x in f:
    l = x[:-1].split(" ")
    sens = "None"
    last = int(l[0])
    test = True
    for e in [int(ele) for ele in l[1:]]:
        if sens == "None":
            if e > last and abs(e - last) <= 3:
                sens = "Croissant"
            elif e < last and abs(e - last) <= 3:
                sens = "Decroissant"
            else:
                test = False
        elif sens == "Decroissant":
            if e >= last:
                test = False
            elif abs(e - last) > 3:
                test = False
        elif sens == "Croissant":
            if e <= last:
                test = False
            elif abs(e - last) > 3:
                test = False
        last = e

    if test:
        print(x, end="")
        somme += 1  
    

print(somme)